framework/test/test_app/app.js

```diff
@@ -28,44 +25,28 @@ const dummyBuildVersion = '#buildVersion';
 const appConfig = {
 	version: '3.0.0',
 	protocolVersion: '2.0',
+	label: 'lisk-devnet',
 	lastCommitId: dummyLastCommitId,
 	buildVersion: dummyBuildVersion,
 	ipc: {
 		enabled: true,
 	},
 };
 
-// Support for PROTOCOL_VERSION only for tests
-if (process.env.NODE_ENV === 'test' && process.env.PROTOCOL_VERSION) {
-	appConfig.protocolVersion = process.env.PROTOCOL_VERSION;
-}
-
 const network = process.env.LISK_NETWORK || 'devnet';
 
 try {
-	// TODO: I would convert config.json to .JS
-	configurator.loadConfig(appConfig);
-	configurator.loadConfigFile(
-		path.resolve(__dirname, `../fixtures/config/${network}/config`),
-	);
+	// eslint-disable-next-line import/no-dynamic-require,global-require
+	const config = require(`../fixtures/config/${network}/config`);
 	// eslint-disable-next-line import/no-dynamic-require,global-require
 	const genesisBlock = require(`../fixtures/config/${network}/genesis_block`);
 
-	if (process.env.CUSTOM_CONFIG_FILE) {
-		configurator.loadConfigFile(path.resolve(process.env.CUSTOM_CONFIG_FILE));
-	}
-
-	const config = configurator.getConfig(appConfig, { failOnInvalidArg: false });
-
-	// Support for PROTOCOL_VERSION only for tests
-	if (process.env.NODE_ENV === 'test' && process.env.PROTOCOL_VERSION) {
-		config.protocolVersion = process.env.PROTOCOL_VERSION;
-	}
-
+	const mergedConfig = {
+		...appConfig,
+		...config,
+	};
```

Comment:
Object spread does not work with deep merge, you have to use some utility function for it.

Reply:
in this case, it will work because appConfig only have first layer, and config will overwrite everything else